src/ResourceManager/Search/Commands.Management.Search/SearchService/NewSearchServiceAdminKeyCommand.cs

```diff
@@ -0,0 +1,101 @@
+//
+// Copyright Microsoft Corporation
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// ----------------------------------------------------------------------------------
+
+using Microsoft.Azure.Commands.Management.Search.Models;
+using Microsoft.Azure.Commands.Management.Search.Properties;
+using Microsoft.Azure.Commands.ResourceManager.Common.ArgumentCompleters;
+using Microsoft.Azure.Management.Internal.Resources.Utilities.Models;
+using Microsoft.Azure.Management.Search.Models;
+using System;
+using System.Management.Automation;
+
+namespace Microsoft.Azure.Commands.Management.Search.SearchService
+{
+    [Cmdlet(VerbsCommon.New, SearchServiceAdminKeyNounStr, SupportsShouldProcess = true), OutputType(typeof(PSSearchAdminKey))]
+    public class NewSearchServiceAdminKeyCommand : SearchServiceBaseCmdlet
+    {
+        [Parameter(
+           Position = 0,
+           Mandatory = true,
+           ValueFromPipeline = true,
+           ParameterSetName = ParentObjectParameterSetName,
+           HelpMessage = InputObjectHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public PSSearchService ParentObject { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ParentResourceIdParameterSetName,
+            HelpMessage = ResourceIdHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string ParentResourceId { get; set; }
+
+        [Parameter(
+            Position = 0,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ResourceNameParameterSetName,
+            HelpMessage = ResourceGroupHelpMessage)]
+        [ResourceGroupCompleter()]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            ParameterSetName = ResourceNameParameterSetName,
+            HelpMessage = ResourceNameHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public string ServiceName { get; set; }
+
+        [Parameter(
+           Mandatory = true,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = KeyKindHelpMessage)]
+        [ValidateNotNullOrEmpty]
+        public AdminKeyKind KeyKind { get; set; }
```

Comment:
Can you wrap AdminKeyKind in a PS model? This will allow you to make breaking changes to the SDK that are mitigated in PowerShell until the breaking change release.